py-polars/polars/expr/expr.py

```diff
@@ -292,20 +292,34 @@ def __array_ufunc__(
         is_custom_ufunc = getattr(ufunc, "signature") is not None  # noqa: B009
         num_expr = sum(isinstance(inp, Expr) for inp in inputs)
         exprs = [
-            (inp, Expr, i) if isinstance(inp, Expr) else (inp, None, i)
+            (inp, True, i) if isinstance(inp, Expr) else (inp, False, i)
             for i, inp in enumerate(inputs)
         ]
+
         if num_expr == 1:
-            root_expr = next(expr[0] for expr in exprs if expr[1] == Expr)
+            root_expr = next(expr[0] for expr in exprs if expr[1])
         else:
-            root_expr = F.struct(expr[0] for expr in exprs if expr[1] == Expr)
+            # We rename all but the first expression in case someone did e.g.
+            # np.divide(pl.col("a"), pl.col("a")); we'll be creating a struct
+            # below, and structs can't have duplicate names.
+            first_renamable_expr = True
+            new_exprs = []
+            for inp, is_actual_expr, index in exprs:
+                if is_actual_expr:
+                    if first_renamable_expr:
+                        first_renamable_expr = False
+                    else:
+                        inp = inp.alias(f"argument_{index}")
+                new_exprs.append((inp, is_actual_expr, index))
+            exprs = new_exprs
+            root_expr = F.struct(expr[0] for expr in exprs if is_actual_expr)
```

Comment:
Instead of `new_exprs` I'd maybe call this `actual_exprs` and then only append to it when `is_actual_expr` and then feed that to `root_expr`'s `F.struct`.

As you have it here, when you build the struct, `is_actual_expr` is just the last value from `exprs` above since it's not in the loop where you unpack and name the tuple.

Reply:
I set a bad precedent because I named my variable exprs when those are really just input args not expressions.